File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/SubarrayDivision/SubarrayDivision.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleTestSolutions.HackerRank.SubarrayDivision
{
    class SubarrayDivision
    {
        public static int birthday(List<int> s, int d, int m)
        {
            int result = 0, count = 0;

            for (int i = 0; i < s.Count; i++)
            {
                if ((i + m - 1) < s.Count)
                {
                    for (int j = i; j < (i + m); j++)
                    {
                        result += s[j];
                    }

                    if (result == d) ++count;

                    result = 0;
                }
            }

            return count;
        }

    }
}

File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/SubarrayDivision/SubarrayDivision.java
import java.util.List;

public class SubarrayDivision {
    public static int birthday(List<Integer> s, int d, int m){
        int result = 0, count = 0;

        for (int i = 0; i < s.size(); i++) {

            if ((i + m - 1) < s.size()) {

                for (int j = i; j < (i + m); j++) {

                    result += s.get(j);

                }

                if (result == d) count++;

                result = 0;
            }
        }
        return result;
    }
}